rust/lance/src/dataset/write.rs

```diff
@@ -111,6 +112,22 @@ impl TryFrom<&str> for WriteMode {
     }
 }
 
+/// Auto cleanup parameters
+#[derive(Debug, Clone)]
+pub struct AutoCleanupParams {
+    pub interval: usize,
+    pub older_than: TimeDelta,
+}
+
+impl Default for AutoCleanupParams {
+    fn default() -> Self {
```

Comment:
@dsgibbons Does it mean the auto cleanup takes effect by default? Is there a switch for this?

Reply:
@yanghua That's right. I do wonder whether enabling destructive actions should be the default. You can set `auto_cleanup: None` in your `WriteParams` if you want to disable this behaviour. A couple of the existing unit tests were altered in this way by this PR to disable auto cleanup.

Reply:
Can we set None as the default value? Users may not know this default behavior and may not expect it.

Reply:
Sounds good to me. Thoughts @wjones127?

Reply:
@wjones127 Any thoughts?

Reply:
@yanghua let's discuss this in a dedicated issue, so the discussion is easier to find. Most people don't go looking in comment on closed PRs.

I've started a discussion here:
https://github.com/lancedb/lance/issues/3854